client/txn.go

```diff
@@ -422,9 +421,16 @@ func (txn *Txn) send(calls ...proto.Call) error {
 		return err
 	}
 
-	if haveEndTxn {
-		// We sent an EndTransaction and it succeeded.
-		txn.finishedOK = true
+	// We sent an EndTransaction and it succeeded. Accept the Transaction
+	// returned from the server.
+	if haveEndTxn && len(calls) > lastIndex {
+		// TestCommitTransactionOnce needs this null check. Why?
+		if endTxnResp.Txn != nil {
+			txn.Proto = *endTxnResp.Txn
```

Comment:
won't this pass through `func (ts *txnSender) Send(ctx context.Context, call proto.Call)`? That's the part that I meant.